Predict the marks obtained based on the Number of 
hours studied and the number of hours slept
FileName: MultipleLinearRegression.ipynb
Dataset: Students02.csv

In [1]:
# Import Pandas for data processing
import pandas as pd

In [2]:
# Read the CSV file
dataset = pd.read_csv('Students02.csv')

In [4]:
# Split into X (Independent) and Y (predicted)
X = dataset.iloc[:, :-1]
Y = dataset.iloc[:, -1]

In [5]:
# Split for rows import train_test_split
from sklearn.model_selection import train_test_split

In [6]:
# Split dataset in train / test set
x_train, x_test, y_train, y_test =     \
train_test_split (X, Y, test_size = 0.3, random_state=12)

In [7]:
# train the Simple Linear Regression
from sklearn.linear_model import LinearRegression

In [8]:
lin_reg = LinearRegression()

In [9]:
# Provide the training Data
lin_reg.fit(x_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [10]:
# predict the results
y_predict = lin_reg.predict(x_test)

In [11]:
# find the R-Squared 
mlr_score = lin_reg.score(x_test, y_test)

In [28]:
# get Coefficient and Intercept
lr_coefficient = lin_reg.coef_
lr_intercept = lin_reg.intercept_
print("Coefficient =" , lr_coefficient)
print("Intercept =" , lr_intercept)

Coefficient = [4.59780591 5.02695322]
Intercept = 0.7917863602510948


Equation of the line
y = Intercept + Coefficient * StudyHours + Coefficient * SleptHours
y = 0.7917863602510948  + 4.59780591 * StudyHours +5.02695322 * SleptHours
this actually tells us that if the student do not study and sleep,
he/she will get 0.7917863602510948 or just 1 mark and its make sense

Calculate the errors using RMSE 

In [13]:
from sklearn.metrics import mean_squared_error
import math

In [29]:
lr_rmse = math.sqrt(mean_squared_error(y_test, y_predict))
print("RMSE = ", lr_rmse)

RMSE =  6.6125162174475
